### PR TITLE
Support drag and drop in Safari (#7420)

### PR DESCRIPTION
**Pull request by aaronraimist**
Fixes https://github.com/vector-im/riot-web/issues/7420

### CODE REVIEW

**Comment by aaronraimist** [comment]
Sorry for the delay. How about this? This does basically the same thing as the shipping version does but with APIs that are available in Safari as well.

File: src/components/structures/RoomView.js [hunk 1]
```diff
@@ -897,18 +897,9 @@ module.exports = React.createClass({
         ev.stopPropagation();
         ev.preventDefault();
 
-        ev.dataTransfer.dropEffect = 'none';
```
Comment (turt2live) [inline src/components/structures/RoomView.js]: Don't we still need this to maintain the state?

Comment (aaronraimist) [reply src/components/structures/RoomView.js]: Not as far as I know. "none" doesn't mean "don't show the overlay", it means whatever you are dragging can't be dropped. https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect

Comment (aaronraimist) [reply src/components/structures/RoomView.js]: onDragLeaveOrEnd sets `draggingFile: false` so the overlay goes away

Comment (aaronraimist) [reply src/components/structures/RoomView.js]: oh hold on

Comment (aaronraimist) [reply src/components/structures/RoomView.js]: oh yeah this is right, if we put that in you loose the + sign by the pointer

**Comment by aaronraimist** [comment]
There is more stuff that I need to fix with drag and drop though so don't merge this yet

**Comment by aaronraimist** [comment]
Never mind I don't have enough time to make this any better. I rewrote AuxPanel.js into an ES6 class and moved this stuff

```
    <div className="mx_RoomView_fileDropTarget">
        <div className="mx_RoomView_fileDropTargetLabel"
          title={_t("Drop File Here")}>
            <TintableSvg src="img/upload-big.svg" width="45" height="59" />
            <br />
            { _t("Drop file here to upload") }
        </div>
    </div>
```
into its own PureComponent class in the hopes that could improve flickering but it does not, it works the exact same. If you want that I can commit it otherwise this PR should be done.

Also TIL there are like 5 other drag and drop PRs open so I'm not sure which one is the right one to merge.

**Comment by jryans** [comment]
From the comments above, this sounds like it's been abandoned, so I don't think there's a need to keep it open in this state, as we can always find the code here again if needed. If you are still interested in pursuing this feature, please discuss with us in #riot-dev:matrix.org to find a good approach forward.

**Pull request closed without merging**